src/dfx/src/lib/extension/manager/list.rs

```diff
@@ -0,0 +1,24 @@
+use super::ExtensionManager;
+use crate::lib::{error::ExtensionError, extension::Extension};
+
+impl ExtensionManager {
+    pub fn list_installed_extensions(&self) -> Result<Vec<Extension>, ExtensionError> {
+        let dir_content = self
+            .dir
+            .read_dir()
+            .map_err(ExtensionError::ExtensionsDirectoryIsNotReadable)?;
```

Comment:
This will include the path in the error
```suggestion
        let dir_content = dfx_core::fs::read_dir(&self.dir).map_err(...)?;
```